robocop/checkers/spacing.py

```diff
@@ -0,0 +1,119 @@
+"""
+Spacing checkers
+"""
+from robot.parsing.model.blocks import TestCase, Keyword
+from robot.parsing.model.statements import EmptyLine, Comment
+from robocop.checkers import RawFileChecker, VisitorChecker
+from robocop.rules import RuleSeverity
+
+
+def register(linter):
+    linter.register_checker(InvalidSpacingChecker(linter))
+    linter.register_checker(MissingTrailingBlankLineChecker(linter))
+
+
+class InvalidSpacingChecker(RawFileChecker):
+    """ Checker for invalid spacing. """
+    rules = {
+        "1001": (
+            "trailing-whitespace",
+            "Trailing whitespace at the end of line",
+            RuleSeverity.WARNING
+        )
+    }
+
+    def check_line(self, line, lineno):
+        stripped_line = line.rstrip('\n')
+        if stripped_line:
+            if stripped_line[-1] == ' ':
+                self.report("trailing-whitespace", lineno=lineno, col=len(stripped_line))
+
+
+class MissingTrailingBlankLineChecker(VisitorChecker):
+    """ Checker for invalid spacing. """
+    rules = {
+        "1002": (
+            "missing-trailing-blank-line",
+            "Missing trailing blank line at the end of file",
+            RuleSeverity.WARNING
+        ),
+        "1003": (
+            "empty-lines-between-sections",
+            "Invalid number of empty lines between sections (%d/%d)",
+            RuleSeverity.WARNING,
+            ("empty_lines_between_sections", "empty_lines_between_sections", int)
```

Comment:
Configurables are in format of ("public name", "internal_name", type). Since if we're configuring we're already giving rule name or id we end up with this:
``
--configure empty-lines-between-sections:empty_lines_between_sections:1
```
How about making public name bit shorter? ie
```
("empty-lines", "empty_lines_between_sections", int)
```
```
--configure empty-lines-between-sections:empty-lines:2
```
(or different name). Public name needs to be only unique within rule(as we can have multiple configurables in each rule too) while private name within whole checker

Reply:
Do we really need to provide `internal_name` since we have the mapping always in the messages dictionary? Maybe let's make it easier and just give `public_name` and the internal one will be resolved automatically? Or am I missing something here?

Reply:
Internal name is later referring to attribute in checker, that's why internal_name should be unique. It could be resolved automatically only if we give some rule to it. For example <name_of_the_rule>+<public_name> = internal name. Otherwise we would need to make public name unique which would make this impossible:
```
        "0504": (
            "too-long-test-case",
            "Test case is too long (%d/%d)",
            RuleSeverity.WARNING,
            ('max_len', 'testcase_max_len', int)
```
(I can use "max_len" for all lengths rules in given checker thanks to this)

How it works in general:
Class that is handling configuration finds what rule we want to configure and then call this method to find our tuple (public, private, type) in rule:
```
configurable = msg.get_configurable(param)
```
then we're calling configure method from checker directly:
```
checker.configure(configurable[1], configurable[2](value))
```
As you can see we're using internal_name here. And configure method is updating attributes directly:
```
    def configure(self, param, value):
        self.__dict__[param] = value
```

Reply:
Ok, now I see. We can make the public_name also unique but then they would need to be longer to maintain uniqueness. I think the solution we have is better, but we need to provide the name of the rule also. It's fine.